cmux.go

```diff
@@ -103,22 +107,21 @@ func (m *cMux) Serve() error {
 			continue
 		}
 
-		go m.serve(c)
+		wg.Add(1)
+		go m.serve(c, &wg)
 	}
 }
 
-func (m *cMux) serve(c net.Conn) {
+func (m *cMux) serve(c net.Conn, wg *sync.WaitGroup) {
+	defer wg.Done()
+
 	muc := newMuxConn(c)
 	for _, sl := range m.sls {
 		for _, s := range sl.ss {
 			matched := s(muc.sniffer())
 			muc.reset()
 			if matched {
-				select {
-				case sl.l.connc <- muc:
-				case <-sl.l.donec:
-					_ = c.Close()
-				}
+				sl.l.connc <- muc
```

Comment:
I have a service that receives a large number of connection requests, and it gets blocked without this select. That's why I have the select there. Let's keep the select as is, and add the waitgroup to unblock your usecase.